src/ethereum/frontier/fork.py

```diff
@@ -768,7 +768,7 @@ def calculate_block_difficulty(
     # See https://github.com/ethereum/go-ethereum/pull/1588
     num_bomb_periods = (int(block_number) // 100000) - 2
     if num_bomb_periods >= 0:
-        difficulty += 2**num_bomb_periods
+        difficulty += Uint(2**num_bomb_periods)
```

Comment:
```python
num_bomb_periods = (block_number // Uint(100000)) - Uint(2)
if num_bomb_periods >= 0:
    difficulty += Uint(2) ** num_bomb_periods
```

Kinda just thinking out loud, but would it be better in your opinion to do the math as `Uint`, or do the conversion at the end?

Reply:
```
difficulty = int(difficulty)
num_bomb_periods = (int(block_number) // 100000) - 2
if num_bomb_periods >= 0:
    difficulty += 2**num_bomb_periods

return max(Uint(difficulty), MINIMUM_DIFFICULTY)
```

is this better?